src/os/src/detail/osTime.c

```diff
@@ -379,26 +386,45 @@ int64_t convertTimePrecision(int64_t time, int32_t fromPrecision, int32_t toPrec
       return time;  // only to pass windows compilation
     }
   } //end switch fromPrecision
+end_:
+  if (tempResult >= (double)INT64_MAX) return INT64_MAX;
```

Comment:
since int64_t to double conversion will lose precision, tempResult is less than INT64_MAX, greater than 9223372036854775295, the line 390 will return true.
~

Reply:
It does not matter. If tempResult is enough, returning INT64_MAX is ok.